### DISCUSSION TITLE
YouTube Music Support

### DISCUSSION THREAD

**Discussion by ghost** [MissingCore/Music#48] [Ideas]
I like the Ui of this app but could it have support for YouTube music.

**Comment by The-Ladle** [comment]
Spotify support would be nice too

**Comment by cyanChill** [comment]
I just wrote a new document; [`App Goals & Ideology`](https://github.com/MissingCore/Music/blob/main/docs/app-goals-ideology.md) which goes a bit on what I want this app to be along with things I don't want to add.

In short, I don't plan on adding support for YouTube Music or Spotify for the following reasons:
- Costs (I will probably need a server to add support as these services would require an API Key and you don't want to expose that to users).
- There's no official YouTube Music API, in addition, if there is, it'll cost $$.
- Spotify doesn't support streaming with their API, unless you have premium.
- Random legal stuff that'll probably show up.

**Reply by ghost** [reply]
Could it be possible for support for custom extensions to add user made features to the app.

**Reply by cyanChill** [reply]
I'm not sure how that would exactly work (or how that would be implemented with React Native).

**Reply by ghost** [reply]
I guess it would work kind of in the same way Tampermonkey dose.

**Reply by cyanChill** [reply]
If we go the route of allowing the user to upload scripts, I don't think much can be done with them (nor do I think it would work?). You would probably need to generate some UI to interact with your feature and even if we expose some functions to this script, it'll be pretty limited. In addition, the file will probably be difficult to maintain.

There's also the question if the method of audio streaming would be compatible with the media player included with Android.

At this point, it would probably be better to create a fork and implement your feature in that.
